tests/zugferd2/Tests/ReaderTest.php

```diff
@@ -44,7 +44,7 @@ public function dataProvider(): array
 
         $buffer = [];
         foreach ($finder as $file) {
-            $buffer[$file->getFilename()] = [$file];
+            yield [$file];
```

Comment:
```suggestion
            yield $file->getFilename() => [$file];
```

Your change to generator is fine. But i would like to have the filename as dataset name. If the test fails, the offending example is easily found.